ui/dashboards/src/components/PanelGroupDialog/PanelGroupDialog.test.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { GridDefinition } from '@perses-dev/core';
+import { screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import produce from 'immer';
+import * as dashboardAppSlice from '../../context/DashboardAppSlice';
+import * as layoutsSlice from '../../context/LayoutsSlice';
+import { renderWithContext } from '../../test';
+import testDashboard from '../../test/testDashboard';
+import PanelGroupDialog from './PanelGroupDialog';
+
+const dashboardApp = {
+  panelDrawer: undefined,
+  openPanelDrawer: jest.fn(),
+  closePanelDrawer: jest.fn(),
+  panelGroupDialog: {
+    groupIndex: undefined,
+  },
+  openPanelGroupDialog: jest.fn(),
+  closePanelGroupDialog: jest.fn(),
+};
+
+const updateLayout = jest.fn();
+jest.spyOn(layoutsSlice, 'useLayouts').mockReturnValue({
+  updateLayout,
+  addItemToLayout: jest.fn(),
+  layouts: testDashboard.spec.layouts,
+});
```

Comment:
Same nit/question as above about avoiding mocking hooks. (And again, fine if not.)